cmake/FindRust.cmake

```
@@ -288,39 +315,43 @@ function(add_rust_library)
     list(APPEND MY_CARGO_ARGS "--target-dir" ${ARGS_BINARY_DIRECTORY})
     list(JOIN MY_CARGO_ARGS " " MY_CARGO_ARGS_STRING)
 
-    # Build the library and generate the c-binding
-    if("${CMAKE_OSX_ARCHITECTURES}" MATCHES "^(arm64;x86_64|x86_64;arm64)$")
+    GetCargoExtraEnv(_cargo_extra_env)
+    GetCargoLocalDirEnv(_cargo_local_dir_env)
+
+    if(RUST_COMPILER_TARGET STREQUAL "universal-apple-darwin")
+        foreach(arch IN LISTS CMAKE_OSX_ARCHITECTURES)
+            message("Making custom command for OSX arch ${arch}")
+            if(arch STREQUAL "arm64")
+                # Convert to rustc's designation
+                set(arch "aarch64")
+            endif()
+            set(this_output "${ARGS_BINARY_DIRECTORY}/${arch}-apple-darwin/${CARGO_BUILD_TYPE}/lib${ARGS_TARGET}.a")
+            list(APPEND lipo_inputs "${this_output}")
```

This is neat.  I'd suggest a little whitespace plus a comment for reading clarity.

```suggestion

            # Record path of the compiled library for lipo to assemble the universal binary, later. 
            set(this_output "${ARGS_BINARY_DIRECTORY}/${arch}-apple-darwin/${CARGO_BUILD_TYPE}/lib${ARGS_TARGET}.a")
            list(APPEND lipo_inputs "${this_output}")

```